### PR TITLE
docs: add blog post on orchestrate tabby with caddy

### PR DESCRIPTION
**Pull request by wsxiaoys**
TAB-558

### CODE REVIEW

**Comment by wsxiaoys** [comment]
cc @icycodes

File: website/blog/2024-03-19-deploy-tabby-with-replicas-behind-reverse-proxy/index.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+---
+authors: [meng]
+tags: [deployment, reverse proxy]
+---
+
+# Deploying Tabby with Replicas and a Reverse Proxy
+
+Welcome to our tutorial on how to set up Tabby, the self-hosted AI coding assistant, with Caddy serving as a reverse proxy (load balancer). This guide assumes that you have a Linux machine with Docker, CUDA drivers, and the [nvidia-container-toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html) already installed.
```
Comment (gyxlucy) [inline website/blog/2024-03-19-deploy-tabby-with-replicas-behind-reverse-proxy/index.md:8]: Add a little bit of motivation here? What's the benefit of this? Why certain users would need it?

File: website/blog/2024-03-19-deploy-tabby-with-replicas-behind-reverse-proxy/index.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+-d '{
+  "language": "python",
+  "segments": {
+    "prefix": "def fib(n):\n    ",
+    "suffix": "\n        return fib(n - 1) + fib(n - 2)"
+  }
+}'
+```
+
+The response should indicate that Tabby is healthy and ready to assist you with your coding tasks.
```
Comment (gyxlucy) [inline website/blog/2024-03-19-deploy-tabby-with-replicas-behind-reverse-proxy/index.md:109]: Add a screenshot of the expected response?

File: website/blog/2024-03-19-deploy-tabby-with-replicas-behind-reverse-proxy/Caddyfile
Comment (gyxlucy) [inline website/blog/2024-03-19-deploy-tabby-with-replicas-behind-reverse-proxy/Caddyfile:1]: This file doesn't seem to be used in the blog? so does docker-compose.yml

Comment (gyxlucy) [inline website/blog/2024-03-19-deploy-tabby-with-replicas-behind-reverse-proxy/Caddyfile:1]: This file doesn't seem to be used in the blog? so does docker-compose.yml

**Review by gyxlucy — CHANGES_REQUESTED** [review]

File: website/blog/2024-03-19-deploy-tabby-with-replicas-behind-reverse-proxy/Caddyfile
Comment (wsxiaoys) [reply website/blog/2024-03-19-deploy-tabby-with-replicas-behind-reverse-proxy/Caddyfile:1]: added as embed files.

File: website/blog/2024-03-19-deploy-tabby-with-replicas-behind-reverse-proxy/index.md (hunk 1 above)
Comment (wsxiaoys) [reply website/blog/2024-03-19-deploy-tabby-with-replicas-behind-reverse-proxy/index.md]: added

File: website/blog/2024-03-19-deploy-tabby-with-replicas-behind-reverse-proxy/index.md (hunk 2 above)
Comment (wsxiaoys) [reply website/blog/2024-03-19-deploy-tabby-with-replicas-behind-reverse-proxy/index.md]: ack

**Review by gyxlucy — APPROVED** [review]
LGTM

**Pull request merged**